src/libssh2_priv.h

```diff
@@ -619,7 +619,7 @@ struct _LIBSSH2_SESSION
      * Or read from server in (eg) KEXDH_INIT (for client mode)
      */
     unsigned char *server_hostkey;
-    uint32_t server_hostkey_len;
+    size_t server_hostkey_len;
```

Comment:
This is unsigned 32 bit data from the wire. I think it would be wise to keep the source size here, It can't be larger anyway!